### DISCUSSION TITLE
Fn.JSON

### DISCUSSION THREAD

**Discussion by lrworth** [jcouyang/dhall-aws-cloudformation#14] [Ideas]
It would be nice to encode arbitrary JSON inside `Fn.If`. We have YAML like these:

```yaml
Fn::If:
  - MyCondition
  -
    - w
    - x
  -
    - y
    - z
```

```yaml
Fn::If:
  - MyCondition
  - A: 1
  - B: 2
```

Currently there's no way to encode this with `Fn.If` because `Fn.Type` does not support objects or arrays. Not sure whether just adding a `Fn.JSON` is a good idea, but if it is then it could replace `Fn.String`.

**Comment by jcouyang** [comment]
Is it possible to create a reference of the object or array in the parameter, then
```dhall
Fn.If "Condition" (Ref "theobject") (Ref "thearray")
```
